ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/ContractsReader.scala

```diff
@@ -304,6 +304,15 @@ private[appendonlydao] sealed class ContractsReader(
             AND #$tree_event_witnessesWhere
           LIMIT 1 -- limit here to guide planner wrt expected number of results
        ),
+       -- no visibility check, as it is used to backfill missing template_id and create_arguments for divulged contracts
```

Comment:
I am guessing this query didn't work before this correction

Reply:
I would be surprised if it did :) But it was never asserted since all tests were probably running with translation cache disabled.

Reply:
It depends on the ledger. Sandbox and public KV ledgers are lazy wrt creating divulgence events (i.e., they don't add the create argument), as they can rely on the create events always to be visible. So for them the query didn't work.

For a PKV ledger and Canton it did work, as they are using divulgence with create arguments.

On a code level this shows in differences wrt how the fields here are used: https://github.com/digital-asset/daml/blob/main/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala#L192

- Sandbox classic: empty list of divulged contracts and None for blinding info (Indexer computes the blinding info)
- KV: empty list of divulged contracts, but pre-computed blinding-info due to dropped fetch nodes
- PKV, Canton, Corda: divulged contracts and pre-computed blinding-info

Reply:
@meiersi-da Thanks for the background info. I believe @mziolekda was pointing to the SQL not being valid at all, not  functional correctness.